File: android-client/src/jp/archilogic/docnext/android/coreview/image/BindQueueItem.java
package jp.archilogic.docnext.android.coreview.image;

import android.graphics.Bitmap;

public class BindQueueItem {
    public int page;
    public int level;
    public int px;
    public int py;
    public boolean isWebp;
    public Bitmap bitmap;
    public int pointer;

    public BindQueueItem( final int page , final int level , final int px , final int py , final Bitmap bitmap ) {
        this.page = page;
        this.level = level;
        this.px = px;
        this.py = py;
        this.bitmap = bitmap;
        isWebp = false;
    }

    public BindQueueItem( final int page , final int level , final int px , final int py , final int pointer ) {
        this.page = page;
        this.level = level;
        this.px = px;
        this.py = py;
        this.pointer = pointer;
        isWebp = true;
    }
}

File: iOS-client/docnext/Classes/Image/BindQueueItem.m
//
//  BindQueueItem.m
//  docnext
//
//  Created by  on 11/10/06.
//  Copyright 2011 Archilogic. All rights reserved.
//

#import "BindQueueItem.h"

@implementation BindQueueItem

@synthesize page;
@synthesize level;
@synthesize px;
@synthesize py;
@synthesize data;

- (id)initWithParam:(int)p_page level:(int)p_level px:(int)p_px py:(int)p_py data:(GLubyte *)p_data {
    self = [super init];

    if (self) {
        self.page = p_page;
        self.level = p_level;
        self.px = p_px;
        self.py = p_py;
        self.data = p_data;
    }
    
    return self;
}

#pragma mark public

+ (BindQueueItem *)itemWithParam:(int)page level:(int)level px:(int)px py:(int)py data:(GLubyte *)data {
    return [[[BindQueueItem alloc] initWithParam:page level:level px:px py:py data:data] autorelease];
}

@end
